tools/downloader/page_parser.py

```
@@ -0,0 +1,385 @@
+            href = link.get('href')
+            text = link.get_text()
+            
+            if not href:
+                continue
+            
+            # デバッグ: すべてのリンクを表示
+            logger.debug("リンク: %s, テキスト: %s", href, text)
+            
+            # 特定のURLパターンを直接チェック
+            if '/reports/' in href and any(re.search(pattern, text) for pattern in YEAR_PATTERNS):
+                if seasonal_report_only and not '/reports/SS' in href:
```

Ruff (0.8.2)

Test for membership should be `not in`

Convert to `not in`

(E713)